### DISCUSSION TITLE
Getting all Object Types in iTHOR

### DISCUSSION THREAD

**Discussion by gzhihongwei** [allenai/ai2thor#1055] [Q&A] [answered]
Hi there,

I was trying to compile all of the `objectType`s in iTHOR to get a sense of what objects could be interacted with in the environment. Specifically, I got every object in `RECEPTACLE_OBJECTS`, so both keys and values, and got the unique objects. Here is the script I used:

```python
from functools import reduce
from ai2thor.controller import RECEPTACLE_OBJECTS

objects = reduce(set.union, RECEPTACLE_OBJECTS.values())
objects |= set(RECEPTACLE_OBJECTS.keys())
objects = sorted(objects)

print(len(objects))
```

This prints out 65, which was much less than I was expecting given [this](https://ai2thor.allenai.org/ithor/documentation/objects/object-types) page. When I did a quick sanity check of the `objectType`s on that page,

```javascript
document.querySelectorAll("tr").length
```

I get 127. I know this is an overestimate of the rows, but I'm no where near that amount of objects. Is there somewhere simpler to get all of the different `objectType`s? The only other option I could think of is manually going through each scene, going through the objects in the metadata, then getting the unique `objectType`s, but this would require unity to render every scene (which is not preferable).

**Comment by mattdeitke** [comment, marked as answer]
Maybe try something like:

```python
from ai2thor.controller import Controller
controller = Controller()

kitchens = [f"FloorPlan{i}" for i in range(1, 31)]
living_rooms = [f"FloorPlan{200 + i}" for i in range(1, 31)]
bedrooms = [f"FloorPlan{300 + i}" for i in range(1, 31)]
bathrooms = [f"FloorPlan{400 + i}" for i in range(1, 31)]

scenes = kitchens + living_rooms + bedrooms + bathrooms
object_types = set()
for scene in scenes:
    event = controller.reset(scene)
    for obj in event.metadata["objects"]:
        set.add(obj["objectType"])
print(len(object_types))
```

which iterates through the scenes and prints the object types.
